src/main/java/jenkins/branch/BranchBuildStrategy.java

```diff
@@ -161,12 +162,22 @@ public boolean isAutomaticBuild(@NonNull SCMSource source,
      * @since 2.4.2
      */
     @Restricted(ProtectedExternally.class)
+    public boolean isAutomaticBuild(@NonNull SCMSource source,
+                                             @NonNull SCMHead head,
+                                             @NonNull SCMRevision currRevision,
+                                             @CheckForNull SCMRevision lastBuiltRevision,
+                                             @CheckForNull SCMRevision lastSeenRevision,
+                                             @NonNull TaskListener listener) {
+        return isAutomaticBuild(source, head, currRevision, lastBuiltRevision, lastSeenRevision, listener, null);
+    }
+
     public abstract boolean isAutomaticBuild(@NonNull SCMSource source,
```

Comment:
this pattern of keep adding parameters is getting worst and worst :(
Maybe we could have something such (pseudo code) to avoid breaking backward compat with every new data: 
```
class AutomaticBuild {
  SCMHead head;
  SCMRevision currRevision;
  // all the getters/setters
}
```
or wait to be 17 here and use `record` :)

Reply:
We can use `record` now.

Is this PR still relevant? Is there some proposed caller?